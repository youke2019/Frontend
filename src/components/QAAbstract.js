import React from "react";
import {
  View,
  Text,
  FlatList,
  Image,
  ImageBackground,
  StyleSheet,
  TouchableOpacity,
} from "react-native";
import axios from 'axios';
import { ShadowedTitle } from './ShadowedTitle'
import CommentItem from './CommentItem'

const Tag = (props) =>{
  return(
    <ImageBackground
      style = {styles.tag_style}
      imageStyle={{resizeMode: 'stretch'}}
      source={{uri:'tag'}}
    >
      <Text> {props.text}</Text>
    </ImageBackground>
  )
}

class QAAbstractTitle extends React.Component{
    render () {
      const {tags = ["老师好看吗","怎么考核啊"]} = this.props;
      return(
        <View style = {styles.header_container}>
          <ShadowedTitle text={"问答"}  uri ={"wenda"}/>
          {
            tags.map((tag,index) =>
              <Tag text = {tag} key={index} style = {styles.tag_text}/>)
          }
        </View>
      )
    }
}
export default class QAAbstract extends React.Component{
  state ={
    comments: [],
  }
  componentDidMount () {
  }
  onPressButton =()=>{
    console.log("press")
  }
  render() {

    return(
      <View style = {styles.container}>
        <QAAbstractTitle/>
        <View style = {styles.first_comm}>
          <Text > QA</Text>
        </View>

        <View style = {styles.button_container}>
          <TouchableOpacity
            onPress={this.onPressButton}
            style = {styles.button_touchable}
            activeOpacity={0.3}
          >
            <Text
              style={styles.button_text}
            >全部问答</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}
const styles = StyleSheet.create({
  first_comm:{
    marginLeft: 20,
  },
  button_text:{
    color:"#ff812e",
    textAlign: 'center',
  },
  button_touchable:{
    borderRadius:20,
    paddingHorizontal:50,
    paddingVertical:5,
    backgroundColor:'#200948'
  },
  button_container:{
    marginTop:10,
    flexDirection: 'row',
    justifyContent:'center',
  },
  header_container:{
    flexDirection:'row',
    alignItems:'center',
    flexWrap:'wrap',
  },
  tag_style:{
    paddingHorizontal: 5,
    marginHorizontal:5,
  },
  tag_text:{
    textAlign:"center",
  },
  container:{
    borderTopWidth:15,
    borderTopColor:'whitesmoke',
  }

})