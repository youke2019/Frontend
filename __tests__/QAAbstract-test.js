import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16/build'
import QAAbstract from '../src/components/QAAbstract'

configure({adapter: new Adapter()})

jest.mock("../src/utils/DataRequest.js")

test('renders CommentAbstract component correctly', (done) => {
  const tree = renderer.create(
      <QAAbstract onGotoQuestionPage = {()=>{}}  user_id={"test_id"} course_id={"test_course_id"}  />
  ).toJSON();
  setTimeout(()=>{
    expect(tree).toMatchSnapshot();
    done();
  },1000)
});

describe("<QAAbstract/> component",()=>{
  const wrapper = shallow(
    <QAAbstract onGotoQuestionPage = {()=>{}} user_id={"test_id"} course_id={"test_course_id"} />
  );
  const instance = wrapper.instance();
  console.log(instance)
  it("update when receive props",(done)=>{
    instance.componentDidMount()
    instance.componentWillReceiveProps()
    setTimeout(()=>{done();},1000) /*wait for async to end*/
  })
})