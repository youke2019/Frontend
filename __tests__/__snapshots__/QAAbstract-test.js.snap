// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommentAbstract component correctly 1`] = `
<View
  style={
    Object {
      "borderTopColor": "whitesmoke",
      "borderTopWidth": 15,
      "marginBottom": 10,
      "paddingHorizontal": 27,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginHorizontal": 20,
          "marginVertical": 5,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "position": "relative",
          }
        }
      >
        <Image
          onLoadEnd={[Function]}
          source={
            Object {
              "uri": "wenda",
            }
          }
          style={
            Object {
              "height": 20,
              "resizeMode": "contain",
              "width": 20,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        />
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#bdbdbd",
                "height": 20,
                "justifyContent": "center",
                "resizeMode": "contain",
                "width": 20,
              }
            }
            testID="RNE__Image__placeholder"
          />
        </View>
      </View>
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "position": "relative",
            }
          }
        >
          <Image
            onLoadEnd={[Function]}
            source={
              Object {
                "uri": "yellow_line",
              }
            }
            style={
              Object {
                "height": 10,
                "left": 8,
                "position": "absolute",
                "top": 12,
                "width": 40,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          />
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "height": 10,
                  "justifyContent": "center",
                  "left": 8,
                  "position": "absolute",
                  "top": 12,
                  "width": 40,
                }
              }
              testID="RNE__Image__placeholder"
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "lineHeight": 25,
              "marginLeft": 10,
              "maxWidth": 250,
              "textAlign": "left",
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          问答
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": 100,
          "justifyContent": "center",
          "width": 340,
        }
      }
    >
      <Text>
         有什么想问的，问问上过课的同学吧😋
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "orange",
          "borderRadius": 20,
          "borderWidth": 0.5,
          "opacity": 1,
          "paddingHorizontal": 55,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ff812e",
            "textAlign": "center",
          }
        }
      >
        全部问答
      </Text>
    </View>
  </View>
</View>
`;
